x-pack/plugins/apm/public/components/shared/Links/MachineLearningLinks/MLLink.tsx

```diff
@@ -26,28 +25,36 @@ interface Props {
 }
 
 export function MLLink({ children, path = '', query = {}, external }: Props) {
-  const { core } = useApmPluginContext();
   const location = useLocation();
 
-  const risonQuery: MlRisonData & TimepickerRisonData = getTimepickerRisonData(
-    location.search
-  );
+  const {
+    core,
+    plugins: { ml },
+  } = useApmPluginContext();
 
-  if (query.ml) {
-    risonQuery.ml = query.ml;
+  let jobIds: string[] = [];
+  if (query.ml?.jobIds) {
+    jobIds = query.ml.jobIds;
   }
-
-  const href = url.format({
-    pathname: core.http.basePath.prepend('/app/ml'),
-    hash: `${path}?_g=${rison.encode(
-      risonQuery as RisonValue
-    )}&mlManagement=${rison.encode({ groupIds: ['apm'] })}`,
+  const { time, refreshInterval } = getTimepickerRisonData(location.search);
```

Comment:
I think we can now ditch `getTimepickerRisonData` and simply instead use `useUrlParams`:

```ts
const { urlParams } = useUrlParams();
const { rangeFrom, rangeTo, refreshInterval } = urlParams;

  const mlADLink = useMlHref(ml, core.http.basePath.get(), {
    page: ML_PAGES.ANOMALY_DETECTION_JOBS_MANAGE,
    pageState: {
      jobId: jobIds,
      groupIds: ['apm'],
      globalState: {
        time: { rangeFrom, rangeTo },
        refreshInterval,
      },
    },
  });
```

Reply:
@smith Since you worked on this most recently: sound ok with you?

Reply:
Updated to the use the `useUrlParams` hook here https://github.com/elastic/kibana/pull/78209/commits/c890c97a8f12a9a3cae5176a24e0bf86dfbb20ed